### PR TITLE
Convert terraform example files to be v0.12 compatible

### LINKED ISSUE #463: Error: Incorrect attribute value type deploying on AWS with Terraform
**What happened**:
Error running `terraform plan`.

**What is the expected behavior**:
`terraform plan` should execute without errors.

**How to reproduce the issue**:
```
$ terraform plan

Error: Incorrect attribute value type

  on main.tf line 180, in resource "aws_lb" "control_plane":
 180:   subnets            = ["${local.all_subnets}"]

Inappropriate value for attribute "subnets": element 0: string required.
```

**Anything else we need to know?**
Using files from https://github.com/kubermatic/kubeone/tree/master/examples/terraform/aws

```
$ terraform version
Terraform v0.12.0
+ provider.aws v2.11.0
```
```
$ cat terraform.tfvars 
cluster_name = "demo"
aws_region = "sa-east-1"
```

**Information about the environment**:
KubeOne version (`kubeone version`): 
```
$ kubeone version
{
  "kubeone": {
    "major": "0",
    "minor": "6",
    "gitVersion": "0.6.2",
    "gitCommit": "7b86c3bf654366a64da608c7b710360dd5f5d777",
    "gitTreeState": "",
    "buildDate": "2019-05-13T16:49:50Z",
    "goVersion": "go1.12.5",
    "compiler": "gc",
    "platform": "linux/amd64"
  },
  "machine_controller": {
    "major": "1",
    "minor": "1",
    "gitVersion": "v1.1.5",
    "gitCommit": "",
    "gitTreeState": "",
    "buildDate": "",
    "goVersion": "",
    "compiler": "",
    "platform": "linux/amd64"
  }
}
```

Operating system: 
```
$ uname -a
Linux artorias 4.9.0-9-amd64 #1 SMP Debian 4.9.168-1+deb9u2 (2019-05-13) x86_64 GNU/Linux
```
Provider you're deploying cluster on:
AWS
Operating system you're deploying on:
Linux

### PR DESCRIPTION
**Pull request by eqrx**
Have done that by running `terraform 0.12upgrade` for each example

**What this PR does / why we need it**:
Terraform upstream version is 0.12 and we should support it. v0.12 introduces language breaking changes that make 0.12 and 0.11 incompatible with each other.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #463

**Special notes for your reviewer**:
None
 
**Documentation**:
Rewrote Docs to match version, added terraform version to compatibility matrix

**Release note**:
Since only examples are changes, does this require "action required"?
```release-note
Terraform examples are now written for version 0.12, but will not work with version 0.11 anymore.
```

### CODE REVIEW

File: docs/quickstart-aws.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ As a result, you'll get Kubernetes 1.14.1 High-Available (HA) clusters with thre
 To follow this quick start, you'll need:
 
 * `kubeone` v0.7.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
-* `terraform` v0.11 installed. The binaries for `terraform` can be found on the [Terraform website](https://www.terraform.io/downloads.html)
+* `terraform` v0.12 installed. The binaries for `terraform` can be found on the [Terraform website](https://www.terraform.io/downloads.html)
 
 **Note:** Due to breaking changes made in Terraform v0.12, it's currently not possible to use example Terraform scripts with Terraform v0.12.
```
Comment (xmudrii) [inline docs/quickstart-aws.md]: We should probably remove this note from all files or modify it to note that you can't use Terraform older than v0.12.

Comment (eqrx) [reply docs/quickstart-aws.md]: Oh sorry, overlooked it. I could modify the prerequisite in all examples to something like "terraform v0.12 or later installed. Examples do not work with v0.11 and older" Would that be good?

Comment (xmudrii) [reply docs/quickstart-aws.md]: Yes, that sounds great.

Comment (eqrx) [reply docs/quickstart-aws.md]: Done

Comment (xmudrii) [reply docs/quickstart-aws.md]: Can you just remove the note that says scripts are not compatible with Terraform 0.12? Otherwise looks great. :100:

Comment (eqrx) [reply docs/quickstart-aws.md]: Urg, forgot to push it, sorry :D

**Comment by kron4eg** [comment]
terraform output json has changed, and requires changes in https://github.com/kubermatic/kubeone/blob/master/pkg/terraform/config.go#L41-L57

**Comment by kron4eg** [comment]
there is new target to convert, vsphere just got merged :) please rebase and cover it too.

**Comment by kron4eg** [comment]
I figured, since we going to need to stick to 0.12 version of terraform, we'd need to update compatibility [matrix here](https://github.com/kubermatic/kubeone/blob/master/README.md#kubernetes-versions-compatibility) and indicate which kubeone version is compatible with which terraform version.

**Comment by eqrx** [comment]
I needed to rebase again to match master. If tests fail, rerun them please. They went through the last attempts on second execution :)

File: docs/quickstart-azure.md [hunk 2]
```diff
@@ -12,13 +12,8 @@ three control plane nodes and two worker nodes.
...
 To follow this quick start, you'll need:
 
-* `kubeone` v0.8.0 or newer installed, which can be
-  done by following the `Installing KubeOne` section of [the README][1],
-* `terraform` v0.11 installed. The binaries for `terraform` can be found on the
-  [Terraform website][2]
-
-**Note:** Due to breaking changes made in Terraform v0.12, it's currently not
-possible to use example Terraform scripts with Terraform v0.12.
+* `kubeone` v0.8.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```
Comment (kron4eg) [inline docs/quickstart-azure.md]:
```suggestion
* `kubeone` v0.9.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```

File: docs/quickstart-gce.md [hunk 3]
```diff
@@ -12,13 +12,8 @@ three control plane nodes and two worker nodes.
...
 To follow this quick start, you'll need:
 
-* `kubeone` v0.8.0 or newer installed, which can be done by following the `Installing KubeOne`
-  section of [the
-  README](https://github.com/kubermatic/kubeone/blob/master/README.md),
-* `terraform` v0.11 installed. The binaries for `terraform` can be found on the
-  [Terraform website](https://www.terraform.io/downloads.html)
-
-**Note:** Due to breaking changes made in Terraform v0.12, it's currently not possible to use example Terraform scripts with Terraform v0.12.
+* `kubeone` v0.8.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```
Comment (kron4eg) [inline docs/quickstart-gce.md]:
```suggestion
* `kubeone` v0.9.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```

File: docs/quickstart-aws.md [hunk 4]
```diff
@@ -9,9 +9,7 @@ As a result, you'll get Kubernetes 1.14.2 High-Available (HA) clusters with thre
 To follow this quick start, you'll need:
 
 * `kubeone` v0.8.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```
Comment (kron4eg) [inline docs/quickstart-aws.md]:
```suggestion
* `kubeone` v0.9.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```

File: docs/quickstart-digitalocean.md [hunk 5]
```diff
@@ -9,9 +9,7 @@ As a result, you'll get Kubernetes 1.14.2 High-Available (HA) clusters with thre
 To follow this quick start, you'll need:
 
 * `kubeone` v0.8.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```
Comment (kron4eg) [inline docs/quickstart-digitalocean.md]:
```suggestion
* `kubeone` v0.9.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```

File: docs/quickstart-hetzner.md [hunk 6]
```diff
@@ -9,9 +9,7 @@ As a result, you'll get Kubernetes 1.14.2 High-Available (HA) clusters with thre
 To follow this quick start, you'll need:
 
 * `kubeone` v0.8.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```
Comment (kron4eg) [inline docs/quickstart-hetzner.md]:
```suggestion
* `kubeone` v0.9.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```

File: docs/quickstart-openstack.md [hunk 7]
```diff
@@ -9,9 +9,7 @@ As a result, you'll get Kubernetes 1.14.2 High-Available (HA) clusters with thre
 To follow this quick start, you'll need:
 
 * `kubeone` v0.8.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```
Comment (kron4eg) [inline docs/quickstart-openstack.md]:
```suggestion
* `kubeone` v0.9.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```

File: docs/quickstart-packet.md [hunk 8]
```diff
@@ -12,12 +12,8 @@ three control plane nodes and one worker node (which can be easily scaled).
 
 To follow this quick start, you'll need:
 
-* `kubeone` v0.8.0 or newer installed, which can be done by following the
-  `Installing KubeOne` section of [the README][main_readme],
-* `terraform` v0.11 installed. The binaries for `terraform` can be found on the
-  [Terraform website][terraform_website]
-
-**Note:** Due to breaking changes made in Terraform v0.12, it's currently not possible to use example Terraform scripts with Terraform v0.12.
+* `kubeone` v0.8.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```
Comment (kron4eg) [inline docs/quickstart-packet.md]:
```suggestion
* `kubeone` v0.9.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```

File: docs/quickstart-vsphere.md [hunk 9]
```diff
@@ -12,13 +12,8 @@ three control plane nodes and two worker nodes.
...
 To follow this quick start, you'll need:
 
-* `kubeone` v0.8.0 or newer installed, which can be
-  done by following the `Installing KubeOne` section of [the README][1],
-* `terraform` v0.11 installed. The binaries for `terraform` can be found on the
-  [Terraform website][2]
-
-**Note:** Due to breaking changes made in Terraform v0.12, it's currently not
-possible to use example Terraform scripts with Terraform v0.12.
+* `kubeone` v0.8.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```
Comment (kron4eg) [inline docs/quickstart-vsphere.md]:
```suggestion
* `kubeone` v0.9.0 or newer installed, which can be done by following the `Installing KubeOne` section of [the README](https://github.com/kubermatic/kubeone/blob/master/README.md),
```

File: hack/run_ci_e2e_test.sh [hunk 10]
```diff
@@ -34,7 +34,7 @@ if ! [ -x "$(command -v terraform)" ]; then
   fi
   echo "Installing terraform"
   cd /tmp
-  curl https://releases.hashicorp.com/terraform/0.11.10/terraform_0.11.10_linux_amd64.zip -Lo terraform.zip
+  curl https://releases.hashicorp.com/terraform/0.12.0/terraform_0.12.0_linux_amd64.zip -Lo terraform.zip
```
Comment (kron4eg) [inline hack/run_ci_e2e_test.sh]:
```suggestion
  curl https://releases.hashicorp.com/terraform/0.12.2/terraform_0.12.2_linux_amd64.zip -Lo terraform.zip
```

Comment (kron4eg) [reply hack/run_ci_e2e_test.sh]: but probably I'd make that a variable, sourced from ENV, defaulted to `0.12.2` in case of absence.

**Pull request merged**